src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -80,8 +86,14 @@
           (cond-> (http/json-response* resp)
             count-query (http/add-headers {:count (jdbc/get-result-count count-query)})))))
     (catch com.fasterxml.jackson.core.JsonParseException e
+      (log/errorf e (str "Error executing query '%s' for entity '%s' "
+                         "with paging-options '%s'. Returning a 400 error code.")
+                  (name entity) query paging-options)
       (http/error-response e))
     (catch IllegalArgumentException e
+      (log/errorf e (str "Error executing query '%s' for entity '%s' "
```

Comment:
This change and the one above I added when having difficulty troubleshooting a bug in my found that a test tripped on. My issue was I had the arguments flipped on a function and so a version (:v4) was being passed in where an entity was expected (:nodes). The error was "IllegalArgumentException No matching clause: :v4", but the above code doesn't log anywhere and only returns an error response. The code above now logs the error + stack before returning an error response